fitbenchmarking/fitbenchmarking
# Refactor and document Controller.flag

#### Description of Work

Fixes #572 

This results in a marginally less nicely formatted display of the flags in controllers.html, so I guess it needs to be determined if this is acceptable.  Arguably the correct solution would be to use sphinx autodoc to have a selection of the code documentation (which I don't think is currently the case, although I could be wrong) included and then simply link to the location of `Controllers.flag` in controllers.rst, rather than including it directly.

Also ideally you would have both the `int` `flag` values and the associated `str` meanings in a class level `OrderedDict` within `Controller`, and then dynamically generate the `Controller.flag` docstring from this: this would mean that the the `flag` values and their meanings are always together.  However I decided that implementing this was more hassle than it was worth, particularly as `VALID_FLAGS` is unlikely to change that often.

NB: I haven't checked if a test for the `flags` validation already exists - if not, this should probably be created.

#### Testing Instructions

1. Check that Adding Fitting Software documentation is still acceptable
2. Check that invalid flags can't be passed to a `Controller` object

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?
